### PR TITLE
Add relativity

### PR DESCRIPTION
**Pull request by arjunsavel**
Not yet ready to merge.

TODO:

- [ ] Throw import error if `reboundx` isn't installed

**Motivation**
Looking to address #54; namely, wanting to add relativistic effects to the code to model more massive systems, e.g. binary stars, high-mass planets orbiting close to high-mass stars, systems near black holes, etc. 

I plan on approaching this in two ways: firstly by introducing the change to orbits, and secondly by adding gravitational redshifting (which @ericagol was referencing in the original issue). I've made headway on the former, but the latter might warrant its own PR.

**Description**

Currently, I've made an integration of reboundx into the Python implementation of ReboundOp. If this style of integration is workable, I'll work on a similar PR to `rebound-pymc3`. I might want to create a new orbit that inherits from ReboundOrbit, given that it has a new citation (which I've added to the citation file) — but that might be the only real difference, so not sure if an entirely new orbit is warranted. A demo showing the instantiation of a gr orbit is depicted below. Other forces ("gr_full", "gr_potential") can be passed as well, but the below images only demonstrate the "gr" force.

**Testing**
The local tests run as expected (155 passed, 12 skipped, 1 xfailed), and I plan on writing a few more tests for this feature:
- testing that there be no velocity discrepancy between models if all GR flags are set to 0
- similarly, that there be no difference if GR is turned on but for very low masses, and 
- that there should be a large (pre-calculated) deviation if GR is turned on for a very large mass. 

There is a time penalty for using the GR model — for the ''gr" force, my fiducial model sees a runtime increase of ~60% (29.2 +/- 2.02s to 47.3 +/-1.76s on my system). 

**Relevant screenshots**

I've also attached some images that demonstrate the expected differences.
 
instantiation

The above image shows how users turn on relativistic effects, simply by using binary flags. A PR to the C bindings for ReboundOp would make the extra ReboundOp argument unnecessary.

short_time

longer_time

For my fiducial model (first image), we see the accumulated difference in x-velocity of the modeled planet over the course of these roughly 3,500 days.

### CODE REVIEW

**Comment by dfm** [comment]
This is so awesome! Thanks and let me know as things progress.

**Comment by arjunsavel** [comment]
@dfm ready for review! Not sure why the Windows test fails, but could be related to #92 (there were intermittent Windows failures in previous commits). Happy to write more tests/documentation if this implementation is fine!

**Comment by dfm** [comment]
Sorry about the slow response! Just catching up on things over here. 

This looks really excellent and I'm happy to merge.

One question: would you be willing to hack this a bit to work even if reboundx isn't installed? Perhaps there could be an import error that gets thrown if you try to include relativity, but don't have reboundx installed, otherwise, it works fine with just rebound?

Thanks again for the contribution!!

File: tests/orbits/rebound_test.py [hunk 1]
```diff
@@ -73,6 +73,80 @@ def test_keplerian(orbit):
...
+        m_star=m_star,
+        r_star=1.0,
+        t0=0.5,
+        period=period,
+        ecc=0.1,
+        omega=0.5,
+        Omega=1.0,
+        incl=0.25 * np.pi,
+        m_planet=m_planet,
+        ReboundOp=test_rebound_op,
```
Comment (dfm) [inline tests/orbits/rebound_test.py:104]: What's the point of having this here? Is this so that `rebound_pymc3` doesn't get used?

Comment (arjunsavel) [reply tests/orbits/rebound_test.py:104]: Yep! I took inspiration from the [`theano_ops` testing](https://github.com/exoplanet-dev/exoplanet/blob/1bc753c4b4df3ad1a2217aa7fc29623fdd059ea7/tests/theano_ops/rebound_op_test.py).

**Comment by arjunsavel** [comment]
No worries, and I'll be sure to implement the import error!

**Comment by arjunsavel** [comment]
Sorry that this took so long! I've added the error handling if reboundx isn't installed. The only test that should fail now is `test_full_adapt_sampling`, which I think has been fixed on main.

**Comment by dfm** [comment]
@arjunsavel: Some how I totally missed your update here - I'm so sorry!! I've just fixed some merge conflicts and once the tests pass, I'll merge. Thanks and again I'm sorry for being so slow to get back to this!

**Comment by dfm** [comment]
Take a look at #133 for an updated version of this PR.

**Pull request merged**